File: Storage/Entities/Comment.cs
namespace Storage.Entities;

public class Comment
{
    public Guid Id { get; set; }
    public Guid CommentId { get; set; }
    public Guid ProductId { get; set; }
    public List<Comment>? Comments { get; set; }
    public required string Name { get; set; }
    public required string Phone { get; set; }
    public required string Text { get; set; }
    public int Rating { get; set; }
    public int Like { get; set; }
    public int Dislike { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public Product? Product { get; set; }
    public Guid? UserId { get; set; }
}

File: Angular/client-app/src/app/models/comment.ts
export interface Comment {
    id: string;
    commentId: string;
    productId: string;
    comments: Comment[] | null;
    name: string;
    phone: string;
    text: string;
    rating: number;
    like: number;
    dislike: number;
    createdAt: Date;
    userId: string | null;
}